framework/test/unit/modules/interoperability/mainchain/module.spec.ts

```diff
@@ -302,375 +356,532 @@ describe('initGenesisState', () => {
+			// doesn't matter what it throw it this stage
+			// check base_interoperability_module.spec.ts for _verifyChannelData relevant tests
+			await expect(interopMod.initGenesisState(context)).rejects.toThrow();
 
-			it('should throw if chainInfo.chainData.status===TERMINATED exists but no terminateStateAccount', async () => {
-				const context = createInitGenesisStateContext(
-					{
-						...genesisInteroperability,
-						chainInfos: [
-							{
-								...chainInfo,
-								chainData: {
-									...chainData,
-									status: ChainStatus.TERMINATED,
-									lastCertificate: {
-										...lastCertificate,
-										validatorsHash: computeValidatorsHash(activeValidators, certificateThreshold),
-									},
-								},
-								chainValidators: {
-									activeValidators,
-									certificateThreshold,
-								},
+			expect(interopMod['_verifyChannelData']).toHaveBeenCalled();
+		});
+
+		it('should check that _verifyChainValidators is called from _verifyChainInfos', async () => {
+			jest.spyOn(interopMod, '_verifyChainValidators' as any);
+
+			const context = createInitGenesisStateContext(
+				{
+					...genesisInteroperability,
+					chainInfos: [
+						{
+							...chainInfo,
+							chainValidators: {
+								...chainValidators,
 							},
-						],
-						// No terminatedStateAccount
-						terminatedStateAccounts: [],
-					},
-					params,
-				);
+						},
+					],
+				},
+				params,
+			);
 
-				await expect(interopMod.initGenesisState(context)).rejects.toThrow(
-					`For each chainInfo with status terminated there should be a corresponding entry in terminatedStateAccounts.`,
-				);
-			});
+			// doesn't matter what it throw it this stage
+			// check base_interoperability_module.spec.ts for _verifyChainValidators relevant tests
+			await expect(interopMod.initGenesisState(context)).rejects.toThrow();
 
-			it('should throw if there is an entry in terminateStateAccounts for a chainID that is ACTIVE in chainInfos', async () => {
-				const context = createInitGenesisStateContext(
-					{
-						...genesisInteroperability,
-						chainInfos: [
-							{
-								...chainInfo,
-								chainData: {
-									...chainData,
-									status: ChainStatus.ACTIVE,
-									lastCertificate: {
-										...lastCertificate,
-										validatorsHash: computeValidatorsHash(activeValidators, certificateThreshold),
-									},
-								},
-								chainValidators: {
-									activeValidators,
-									certificateThreshold,
+			expect(interopMod['_verifyChainValidators']).toHaveBeenCalled();
+		});
+	});
+
+	it(`should call _verifyTerminatedStateAccounts from initGenesisState`, async () => {
+		jest.spyOn(interopMod, '_verifyTerminatedStateAccounts' as any);
+
+		await interopMod.initGenesisState(contextWithValidValidatorsHash);
+		expect(interopMod['_verifyTerminatedStateAccounts']).toHaveBeenCalledTimes(1);
+	});
+
+	// test internals of verifyTerminatedStateAccounts
+	describe('verifyTerminatedStateAccounts', () => {
```

Comment:
desc should match the name of the method
```suggestion
	describe('_verifyTerminatedStateAccounts', () => {
```